client/src/auth/index.jsx

```diff
@@ -1,23 +1,21 @@
-import React from 'react';
-import isAuth from './isAuth';
-import { Route, Redirect } from 'react-router-dom';
+import React from "react";
+import isAuth from "./isAuth";
+import { Route, Redirect } from "react-router-dom";
 
 const PrivateRoute = ({ component: Component, ...rest }) => {
   return (
     <Route
       {...rest}
-      component={props => {
-        return isAuth ? (
+      render={props => {
+        isAuth(state => {
+          if (!state) {
+            rest.clearUserInfo();
+          }
+        });
+        return (
```

Comment:
@aminalakhsham This logic might be better off inside componentDidMount() (so you would have to change this into a class component), so that rendering the component doesn't have functions being called as side effects.
That would avoid situations where unrelated prop changes cause isAuth() to be called again and again (because of the component being re-rendered)